.github/workflows/php.yml

```diff
@@ -0,0 +1,98 @@
+---
+name: PHP
+
+on:
+  # See: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch
+  workflow_dispatch:
+
+  # See: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#push
+  push:
+    branches:
+      - '*'         # matches every branch that doesn't contain a '/'
+      - '*/*'       # matches every branch containing a single '/'
+      - '**'        # matches every branch
+    tags:
+      - "*"
+
+  # See: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+
+env:
+  EXTRA_PHP_EXTENSIONS: intl bcmath zip pcntl exif curl gd pdo_pgsql pdo_mysql pdo_sqlite
+  DOCKER_PHP_EXTENSION_INSTALLER_VERSION: 2.2.2
+  COMPOSER_VERSION: 2.7.1
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+
+      # See: https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs
+      matrix:
+        php_version:
+          - 8.2
+          - 8.3
+
+    container:
+      image: php:${{ matrix.php_version }}-cli
+
+    # See: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-concurrency-and-the-default-behavior
+    concurrency:
+      group: php-test-${{ github.ref }}-${{ matrix.php_version }}
+      cancel-in-progress: false
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+
+      # NOTE: we lint PHP files before composer to avoid wasting time
+      # linting everything in vendor/
+      - name: Lint PHP files
```

Comment:
Would probably be good to split this out to a separate "lint / prettier" workflow to enforce code formatting.

That was something that made the recent security patches troublesome.

Reply:
I'm fine with that - is there any specific `prettier` we want to use? 

This step only check them for being syntax correct PHP, it doesn't care about the structure or style itself. 

I put it here, very early in the job flow, to quickly get feedback before doing any other heavy lifting, but should be easy to move to a different job.

Reply:
Ah, okay, so it's not so much a lint step, but a validation step before doing more expensive checks? I'd say fine to leave in here then

Reply:
Yep, let me change it to use long form flag for more readability - it's mostly there to catch bad syntax or cross-PHP version incompatibilities (e.g. new functions/methods/classes)

```
--syntax-check / -l Syntax check only (lint)
```

Reply:
I added https://laravel.com/docs/10.x/pint as a different job - I couldn't figure out what style guide PF uses, seems like "none", so picked Laravels own formatter :)

Reply:
The PR show-casing pint applied to Pixelfed is here: https://github.com/pixelfed/pixelfed/pull/4922